js/customize-object-selector-component.js

```diff
@@ -85,9 +85,11 @@ wp.customize.ObjectSelectorComponent = (function( api, $ ) {
 						}
 					},
 					templateResult: function( data ) {
+						data.disable_edit = true;
 						return $.trim( component.select2_result_template( data ) );
 					},
 					templateSelection: function( data ) {
+						data.disable_edit = ! data.id;
```

Comment:
The `id` is not defined when the template is for the “searching” message or for the “no results found” message?